### ISSUE TITLE
[BUG] Access requests not showing in app

### ISSUE THREAD

**Issue by fidoly** [microsoft/powerapps-tools#355]
After requesting access and either getting a pending or approved request, the My requests list is empty.  Works the same in the iPhone or any browser (PC & Mac).
The requests are being recorded in the SharePoint list OK.

**Comment by stuartridout** [comment]
@fidoly did you remove and re-add the data sources in the Power App when you installed it?

**Comment by fidoly** [comment]
Hi Stuart,

Yes I did.   Everything works as expected except showing the user own reservations.  I suspect it is the ID  column because PowerBI is reporting not finding the ID column in the BAR_request list.

Can you let me know if the ID column is present in the list in a working installation?

Many thanks,
Phil.

**Comment by stuartridout** [comment]
@fidoly - This is because you are logging in with a different email address to the one in your Office 365 profile.  This has now been resolved for your tenant and has been shared with the developer as a consideration for future versions.

**Comment by alwayslearning2** [comment]
@stuartridout I'm having the same issue as fidoly. I followed the instructions and re-added the data sources in the Power App, used my organization's account to create the Power App and create requests (for testing), and don't see any items under My Requests. I can see the items in the SharePoint list.

**Comment by stuartridout** [comment]
@alwayslearning2 - do you have a different identity that you log in compared with your normal email address?

**Comment by alwayslearning2** [comment]
@stuartridout yes, I do. What's the best way to send you my organization's identity? I'm new to GitHub and don't know if I should just post it here. Thank you!

**Comment by pa1-gulati** [comment]
@alwayslearning2  this happens when your email alias doesn't match the O365 identity. We are working on an updated version that accounts for this. Hope to have an update for you soon.

**Comment by alwayslearning2** [comment]
@pa1-gulati Thank you for the update.

**Comment by cristianoag** [comment]
Hello. For those of you looking into a workaround for this bug while the official fix is not ready. Just follow the steps below and the application will work ok for the situations where the UPN attribute value is different from e-mail address in Azure Active Directory.

	1. Create a new column to store the text value of the UPN.
		a. Open SharePoint Online site where the lists were created. In the BAR_Requests list, create an additional single line of text column called RequestorUPN.
	2. Change the Power Automate BARCreateRequests to store the new value
		a. Open the Power Automate website and edit the BARCreateRequests flow. 
		b. Below Get Manager (V2), expand Scope-ManagerExists and then ForEachRequests. Expand ExistingRequestsBelowThreshold. Expand CreateRequests. In the Body area, create the RequestorUPN field in the JSON and store the UserEmail value. 
		c. Expand CreateRequestsOverCapacityFalse. In the Body area, create the RequestorUPN field in the JSON and store the UserEmail value. 
		d. Below Get Manager (V2), expand Scope-NoManager and then ForEachRequests2. Expand ExistingRequestsBelowThreshold 2. Expand CreateRequests 2. In the Body area, create the RequestorUPN field in the JSON and store the UserEmail value. 
		e. Expand CreateRequestsOverCapacityFalse 2. In the Body area, create the RequestorUPN field in the JSON and store the UserEmail value. 
		f. Ensure there are no syntax errors in the body JSON and save the flow.
	3. Create a new requests in the app and check if the column RequestorUPN was updated with the UPN value.
	4. Change three filter in the Building Access app.
		a. Open PowerApps and edit the Building Access app.
		b. Select View, Data Sources. Right click the BAR_Requests connection and select refresh.
		c. On the tree view, click screens and select "Screen My Requests", change to the OnVisible event and perform the change below:

		From:
		// Get Users requests
		ClearCollect(colUserRequests,Filter(BAR_Requests,DateValue >= varTodayFormated,Requestor.Email=varUser.Email));
		To:
		// Get Users requests
		ClearCollect(colUserRequests,Filter(BAR_Requests,DateValue >= varTodayFormated,RequestorUPN=varUser.Email));
	
		d. On the tree view, click screens and expand "Screen Availability", select DatePicker. Change to the OnChange event and perform the change below:
	
		From:
		// Check user dosent have any requests already pending for today
		ClearCollect(colUserReuqestsToday,Filter(BAR_Requests,Requestor.Email = varUser.Email,DateValue = varSelectedDateString ,Active = 1)) ;
		To:
		// Check user dosent have any requests already pending for today
		ClearCollect(colUserReuqestsToday,Filter(BAR_Requests,RequestorUPN = varUser.Email,DateValue = varSelectedDateString ,Active = 1)) ;
	
		e. On the tree view, click screens and select "Screen Building". Change to the OnVisible event and perform the change below:
	
		From:
		// Get last 3 recently requested buildings by user
		ClearCollect(colDistinctBuildingIds, FirstN( Distinct( SortByColumns(  Filter(BAR_Requests, Requestor.Email = varUser.Email),"Created",Descending),BuildingID).Result ,3));
		To:
		// Get last 3 recently requested buildings by user
		ClearCollect(colDistinctBuildingIds, FirstN( Distinct( SortByColumns(  Filter(BAR_Requests, RequestorUPN = varUser.Email),"Created",Descending),BuildingID).Result ,3));
	
	5. Save and Publish the application.

**Comment by alwayslearning2** [comment]
@cristianoag  Awesome, this worked! Thank you!

**Comment by DB-LEC** [comment]
@cristianoag this fix has almost worked except when I create a new booking then when it transfers to the request confirmation page it comes up with a json error and that page is blank but it is populating everything else as it should be. Do you have any suggestions. Having looked at the Screen Request Confirmation form and looked at errors under formulas it shows this

![ErrorCodes](https://user-images.githubusercontent.com/68995394/89636816-20cf1d00-d8a1-11ea-9dab-74b19f9d10a0.png)


It does not seem to able to "find" the user.

**Comment by DB-LEC** [comment]
I have resolved all the issues so far, but what the above "fix" didn't mention was that you have to do the same with the Approver and use their UPN otherwise it will not populate the approvers list. It is effectively the same principle as above but do message me if you need more information.

**Comment by cristianoag** [comment]
@DB-LEC  is correct. That was an attempt to quick fix the issue while a new fixed version was not published. Anyway, I got access to a fixed branch and implemented two additional features that may be interesting for you all. (1.11) has the ability to override the approval process for specific spaces and redirect it to dedicated approvers (1.12) has the additional capability to present questions per building. Please be aware those changes are not official and made on top of beta code. Use at your own risk. I made my repository open for community contribution so we can expand the solution, please fell free to contribute. Code is available here https://github.com/cristianoag/BuildingAccess

**Comment by DB-LEC** [comment]
> @DB-LEC is correct. That was an attempt to quick fix the issue while a new fixed version was not published. Anyway, I got access to a fixed branch and implemented two additional features that may be interesting for you all. (1.11) has the ability to override the approval process for specific spaces and redirect it to dedicated approvers (1.12) has the additional capability to present questions per building. Please be aware those changes are not official and made on top of beta code. Use at your own risk. I made my repository open for community contribution so we can expand the solution, please fell free to contribute.
> 
> Details on the implementations are available on the following links:
> 
> Per Space Approval
> https://techpeanuts.wordpress.com/2020/08/11/covid19-building-access-app-per-space-approval/
> https://github.com/cristianoag/BuildingAccess/releases/tag/1.11
> 
> Questions per Building + Per Space Approval
> https://techpeanuts.wordpress.com/2020/08/12/covid19-building-access-app-questions-per-building/
> https://github.com/cristianoag/BuildingAccess/releases/tag/v1.12
> 
> Code is available here https://github.com/cristianoag/BuildingAccess

@christianoag does your v1.11 allow us to specify a users "Desk Booking Approver" as opposed to their Manager in AD. Some of our "Managers" are "to busy" to bother approving requests and want to delegate it to another person. If we change the Manager Field in AD this breaks the Phone Book and other Reporting Tools we have. Any help will be appreciated.

**Comment by aakashpatel660** [comment]
Hello, 

I am new to this app and when I imported all the components and when I open the Building Access app in edit mode I can see too many errors. 

 First is that on Screen Building for htmlEligibilityCriteria htmltext it throws the error for the if statement. If(!IsBlank(First(colSelectedBuilding).KeyEligibilityCriteria),First(colSelectedBuilding).KeyEligibilityCriteria,If(CountRows(colSelectedBuilding)=0,
varString.NotSelectBuildingEligibility,varString.EligibilityCriteriaNotAvailable))

When i go and take a look at the colselectedbuilding there is not value. How can i load the value into the collection bec in my BAR_Building Sharepoint list i have value. 

can someone help with this app? 

Thank you so much

**Comment by cristianoag** [comment]
This is a known issue and it is caused by the lack of initialization for the colSelectedBuilding when the building list component is loaded for the first time. To fix it please follow the steps below:

- Open power apps.
- Edit Building Access app.
- Select the Screen Building screen.
- Change to the OnVisible event.
- At the end of the command block, insert:

 ClearCollect(colSelectedBuilding,First(colBuildings));
 Clear(colSelectedBuilding);

- Save and publish.

Please note that the last line of the existent code block doesn’t have a ;, please include it before paste the lines of code above as all lines need ; at the end.

**Comment by aakashpatel660** [comment]
Hello, @cristianoag Thank you so much for your response and you solution did fix my issue. I just have another quick question. When I try to play the Building Access app and place a new request and it asks me to Search for the building. When I search for the building it does not populate any building. As you can see in the below picture. 

Thank you so much for your help.

![image](https://user-images.githubusercontent.com/59940632/91625540-2a1a4980-e976-11ea-964f-272e5db7debb.png)


![image](https://user-images.githubusercontent.com/59940632/91625542-31d9ee00-e976-11ea-97f1-43924a8b1b76.png)

**Comment by cristianoag** [comment]
You need to publish the building. It’s status is draft.

________________________________

**Comment by aakashpatel660** [comment]
I want to have that every time when a new request is placed it has to go through manager approval cannot be a auto-approved. I saw that it works on the threshold so can you please assist me that how can I set so every time when a new request is placed it will go through manager approval.

**Comment by cristianoag** [comment]
Change the approval threshold to 0. All request will require approval.

________________________________

**Comment by cristianoag** [comment]
> > @DB-LEC is correct. That was an attempt to quick fix the issue while a new fixed version was not published. Anyway, I got access to a fixed branch and implemented two additional features that may be interesting for you all. (1.11) has the ability to override the approval process for specific spaces and redirect it to dedicated approvers (1.12) has the additional capability to present questions per building. Please be aware those changes are not official and made on top of beta code. Use at your own risk. I made my repository open for community contribution so we can expand the solution, please fell free to contribute.
> > Details on the implementations are available on the following links:
> > Per Space Approval
> > https://techpeanuts.wordpress.com/2020/08/11/covid19-building-access-app-per-space-approval/
> > https://github.com/cristianoag/BuildingAccess/releases/tag/1.11
> > Questions per Building + Per Space Approval
> > https://techpeanuts.wordpress.com/2020/08/12/covid19-building-access-app-questions-per-building/
> > https://github.com/cristianoag/BuildingAccess/releases/tag/v1.12
> > Code is available here https://github.com/cristianoag/BuildingAccess
> 
> @christianoag does your v1.11 allow us to specify a users "Desk Booking Approver" as opposed to their Manager in AD. Some of our "Managers" are "to busy" to bother approving requests and want to delegate it to another person. If we change the Manager Field in AD this breaks the Phone Book and other Reporting Tools we have. Any help will be appreciated.

Yes. V1.12 allows override the default approver (manager) by specific users (per space).

**Comment by aakashpatel660** [comment]
Is it the same as an Occupancy threshold in BAR_Buildings list?

**Comment by cristianoag** [comment]
Edit the building where you want all requests approved by the manager using the Building Access app. Change the threshold to 0. Save and be happy. :)

________________________________

**Comment by aakashpatel660** [comment]
Hello 

Auto-Approval Threshold is already set to 0. 

![image](https://user-images.githubusercontent.com/59940632/91641542-16f98f00-e9f3-11ea-9df9-3357d5b987e6.png)

**Comment by cristianoag** [comment]
So all requests for spaces in this building will require manager approval. None will be auto-approved.

________________________________

**Comment by aakashpatel660** [comment]
Thank you so much for your all help and quick response.

**Comment by cristianoag** [comment]
> > @DB-LEC is correct. That was an attempt to quick fix the issue while a new fixed version was not published. Anyway, I got access to a fixed branch and implemented two additional features that may be interesting for you all. (1.11) has the ability to override the approval process for specific spaces and redirect it to dedicated approvers (1.12) has the additional capability to present questions per building. Please be aware those changes are not official and made on top of beta code. Use at your own risk. I made my repository open for community contribution so we can expand the solution, please fell free to contribute.
> > Details on the implementations are available on the following links:
> > Per Space Approval
> > https://techpeanuts.wordpress.com/2020/08/11/covid19-building-access-app-per-space-approval/
> > https://github.com/cristianoag/BuildingAccess/releases/tag/1.11
> > Questions per Building + Per Space Approval
> > https://techpeanuts.wordpress.com/2020/08/12/covid19-building-access-app-questions-per-building/
> > https://github.com/cristianoag/BuildingAccess/releases/tag/v1.12
> > Code is available here https://github.com/cristianoag/BuildingAccess
> 
> @christianoag does your v1.11 allow us to specify a users "Desk Booking Approver" as opposed to their Manager in AD. Some of our "Managers" are "to busy" to bother approving requests and want to delegate it to another person. If we change the Manager Field in AD this breaks the Phone Book and other Reporting Tools we have. Any help will be appreciated.

Please use this package https://1drv.ms/u/s!AnR50DmsM8Jwl9kWzDgeuBgB40XJxQ?e=8LZw4R

**Comment by aakashpatel660** [comment]
Hello @cristianoag 

just a quick question for you, Building Security App when a user comes someone will check-in or check-out the user but if the user goes out for a lunch and they did check out how they can do check-in back when the user comes back into a building? 

Thank you

**Comment by cristianoag** [comment]
> Hello @cristianoag
> 
> just a quick question for you, Building Security App when a user comes someone will check-in or check-out the user but if the user goes out for a lunch and they did check out how they can do check-in back when the user comes back into a building?
> 
> Thank you

Use the version below and create two reservations for the same day for the same building:

Updated Package (v1.15)
https://1drv.ms/u/s!AnR50DmsM8Jwl9lcnq1teq6ulTgSNw?e=ifkpxg

**Comment by Lan-BZ** [comment]
Access the desk booking, select my requests but nothing showed up. My requests page is empty.
Please help. Thanks
